gfx/layers/wr/WebrenderPaintedLayer.cpp

```diff
@@ -66,8 +66,39 @@ WebRenderPaintedLayer::RenderLayer(wrstate* aWRState)
   } else {
       clip = rect;
   }
+
+  WRImageMask* mask = nullptr;
+  WRImageMask imageMask;
+  Layer* maskLayer = GetMaskLayer();
+
+  if (maskLayer) {
+    RefPtr<SourceSurface> surface = WebRenderLayer::ToWebRenderLayer(maskLayer)->GetAsSourceSurface();
+    if (surface) {
```

Comment:
can we share this huge block of text? seems rather similar to the one in `WebrenderColorLayer.cpp`